src/app/o/[orgId]/surveys/[surveyId]/page.tsx

```diff
@@ -0,0 +1,45 @@
+'use server';
+
+import BackendApiClient from 'core/api/client/BackendApiClient';
+import { Metadata } from 'next';
+import SurveyForm from 'features/surveys/components/surveyForm/SurveyForm';
+import { ZetkinSurveyExtended } from 'utils/types/zetkin';
+import { FC, ReactElement } from 'react';
+
+type PageProps = {
+  params: {
+    orgId: string;
+    surveyId: string;
+  };
+};
+
+export async function generateMetadata({
+  params,
+}: PageProps): Promise<Metadata> {
+  const { orgId, surveyId } = params;
+  const apiClient = new BackendApiClient({});
+  const survey = await apiClient.get<ZetkinSurveyExtended>(
+    `/api/orgs/${orgId}/surveys/${surveyId}`
+  );
+  return {
+    description: survey.info_text,
+    openGraph: {
+      description: survey.info_text,
+      title: survey.title,
+    },
+    title: survey.title,
+  };
+}
+
+// @ts-expect-error Async support missing
```

Comment:
What is the purpose of this comment?

Reply:
Replaced `Async support missing` with https://nextjs.org/docs/app/building-your-application/configuring/typescript#async-server-component-typescript-error so that the full context is immediately available one click away. We're just a TypeScript major version upgrade away from being able to remove this.